### PR TITLE
Fix wrong ColumnConst.isColumnNullable()

### PR DESCRIPTION
**Pull request by 4ertus2**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Bug Fix

Short description (up to few sentences):
Fix coalesce for ColumnConst with ColumnNullable + related changes.

### CODE REVIEW

**Comment by 4ertus2** [comment]
#5689

**Comment by 4ertus2** [comment]
There're places where we check isColumnNullable() and then static_cast it to ColumnNullable. It's totally wrong cause no one column could have isColumnNullable() returning true. Interface method is used as type checker.

**Comment by alexey-milovidov** [comment]
> There're places where we check isColumnNullable() and then static_cast it to ColumnNullable.

The method is named `isColumnNullable` exactly for this reason (to check that the column is an instance of `ColumnNullable`). You can rename to `isNullable` for your changes to be consistent.

**Comment by 4ertus2** [comment]
PR removes IColumn.isColumnNullable() and IColumn.isColumnConst(). Makes isNullable() and isColumnConst() outside of IColumn and checks old x.isColumnNullable() with checkColumn<ColumnNullable>(x). isNullable() returns true for ColumnNullable and for ColumnConst with nullable inside.

File: dbms/src/Columns/ColumnNullable.cpp [hunk 1]
```diff
@@ -454,13 +454,20 @@ void ColumnNullable::checkConsistency() const
...
         return column;
 
-    if (column->isColumnConst())
+    if (isColumnConst(*column))
         return ColumnConst::create(makeNullable(static_cast<const ColumnConst &>(*column).getDataColumnPtr()), column->size());
 
     return ColumnNullable::create(column, ColumnUInt8::create(column->size(), 0));
 }
 
+bool isNullable(const IColumn & column)
```
Comment (alexey-milovidov) [inline dbms/src/Columns/ColumnNullable.cpp]: Pending LowCardinality.

**Review by alexey-milovidov — CHANGES_REQUESTED** [review]
.

File: dbms/src/Columns/ColumnNullable.cpp (hunk 1 above)
Comment (4ertus2) [reply dbms/src/Columns/ColumnNullable.cpp]: I've add IColumn.isNullble() and overridden it in ColumnConst, ColumnNullable and ColumnLowCardinality.

**Pull request merged**